Brendon/bootcamp page

brendon250891 opened the pull request:
## Bootcamp Landing Page
- Made a landing page for users to view information about courses being offered through bootcamp.
- Based of a mockup i saw a while ago in the docs and files on basecamp.
- Created a course factory to mock some data.
- Used the pipe symbol to act as delimiter to create the list for the course contents so it can be stored as a string in the database.
- Created a course controller for route handling.

![image](https://user-images.githubusercontent.com/23717298/78238853-c1df4d00-7528-11ea-818e-d423bd885aba.png)

![image](https://user-images.githubusercontent.com/23717298/78238992-ea674700-7528-11ea-9a6d-2f7ba8771778.png)

brendon250891 commented:
# Course Changes
- Added description column to $casts as an array.
- This removed the need to complicate the seeding and retrieving of the description by having to mess around with a delimiter.

stevethomas commented on app/Http/Controllers/CourseController.php:
```diff
@@ -0,0 +1,20 @@
+    {
+        $courses = Course::all();
+
+        return view('bootcamp.index', compact('courses'));
```
any reason why this isn't `courses.index`?

stevethomas commented on database/seeds/DatabaseSeeder.php:
```diff
@@ -16,27 +17,28 @@ class DatabaseSeeder extends Seeder
+//
+//            // Seed AssessmentItems
+//            factory(AssessmentItem::class, 2)->create(['assessment_id' => $i]);
+//        }
```
hmm this will probably create a conflict with master once the other branch is merged. Not a lot you can do, will have to resolve conflicts by hand.

stevethomas commented on public/js/app.js:
```diff
@@ -19346,8 +19346,8 @@ window.axios.defaults.headers.common['X-Requested-With'] = 'XMLHttpRequest';
-__webpack_require__(/*! /Users/jerrell/code/bootcamp4/resources/js/app.js */"./resources/js/app.js");
-module.exports = __webpack_require__(/*! /Users/jerrell/code/bootcamp4/resources/css/app.css */"./resources/css/app.css");
+__webpack_require__(/*! /Users/brendon/code/bootcamp4/resources/js/app.js */"./resources/js/app.js");
+module.exports = __webpack_require__(/*! /Users/brendon/code/bootcamp4/resources/css/app.css */"./resources/css/app.css");
```
this file should be .gitignored one of the branches or this will keep happening.

stevethomas commented on routes/web.php:
```diff
@@ -20,3 +20,7 @@
 Route::get('/home', 'HomeController@index')->name('home');
 
+Route::resource('courses', 'CourseController')->names([
+   'index' => 'bootcamp',
```
same again, I would have just called this `courses` as it is the index for courses.

brendon250891 commented:
## Changes Made
- Renamed the route bootcamp to courses
- Moved the courses index.blade.php to the courses directory
- Added a path function to the Course class
- Added public/css/* and public/js/* to my gitignore.
